File: cpp/cpp_1000-/1217.cc
// C++
// leetcode 1217
// https://leetcode.cn/problems/minimum-cost-to-move-chips-to-the-same-position/

class Solution {
public:
    int minCostToMoveChips(vector<int>& position) {
        int cnt = 0, n = position.size();
        for(int i = 0; i < n; i++) {
            cnt += position[i] & 1;
        }
        return min(cnt, n - cnt);
    }
};

File: Go/Go_1000-/1217.go
// Go
// leetcode 1217
// https://leetcode.cn/problems/minimum-cost-to-move-chips-to-the-same-position/

func minCostToMoveChips(position []int) int {
	cnt := 0
	for _, p := range position {
		cnt += p&1
	}
	if len(position) - cnt < cnt {
		return len(position) - cnt
	}
	return cnt
}